lib/rules/one-var.js

```diff
@@ -62,20 +120,46 @@ module.exports = function(context) {
         "FunctionDeclaration": startFunction,
         "FunctionExpression": startFunction,
         "ArrowFunctionExpression": startFunction,
+        "BlockStatement": startBlock,
```

Comment:
`BlockStatement`s aren't the only node type that create a new lexical environment. For example, a lexical declaration in the initializer of a `for` statement isn't available outside. Searching for references to `NewDeclarativeEnvironment` in [the spec](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-newdeclarativeenvironment), the same applies to the `case` block of a `switch` statement. (Note that this list is not exhaustive of all such constructs: `catch` clauses, for example, are already covered by `BlockStatement`; whereas `class` bodies create a new lexical environment but cannot directly contain variable declarations.) @nzakas have I missed any?